docs/cold_reset.md

```diff
@@ -0,0 +1,113 @@
+result in additional reset when set.
+
+###### Reboot
+
+Done with `reboot` in shell or after changing options in sortbootorder. It
+performs warm reset.
+
+###### Reset button
+
+Done by temporary shorting reset pin to the ground. Performs cold reset from
+AGESA's point of view.
+
+###### Power button
+
+By shorting power button pin to the ground for more than 4 seconds platform
+enters S5 state. After second, short press the platform starts up. It is
+considered to be cold reset, but during FCH initialization platform will perform
+another reset (after which PMxC0 is set as in reboot, D18F0x6C as in cold boot).
+This was one of causes for doubled sign of life.
+
+###### FullRst through IO port CF9
+
+Bit 3 of IO port CF9 is:
+
+> **FullRst**. Read-write. Reset: 0. 0=Assert reset signals only. 1=Place system
+> in S5 state for 3 to 5 seconds.
+
+Bits 1 (SysRst) and 2 (RstCmd) need to be set as well. CF9 can be accessed
+through its shadow in ACPI space (PMxC5, at address 0xFED803C5 when MMIO is
+enabled). Performs cold reset and gives time for most peripheral devices to get
+into stable state.
+
+###### ACPI emulated reset button
+
+Done by setting bit 6 (ResetAllAcpi) of PMxC4 (Reset Command). It is described
+as emulating a reset button event, but it does not set UsrReset in PMxC0, while
+physical button does.
+
+###### PCI reset
+
+Also triggered through PMxC4. Reset is done after setting bit 0 (Reset), but
+only if bit 7 (ResetEn) was set. They need to be set with two separate writes.
+Performs "soft PCI reset", which is a warm reset.
+
+### Forcing cold reset from started OS
+
+After flashing new firmware it is highly recommended (sometimes even necessary)
+to perform a cold boot. It is not always an option for remote devices, so tests
+were made to find another way of forcing cold boot path in firmware. Note that
+this cannot guarantee that when any device on board entered an unexpected state
+it can be brought back to defined state without full power cycle.
+
+As these steps are most likely to be run on production software stack options
+like using custom kernel/driver or applying kernel parameters at boot time were
+not considered. Tests were made with Debian 9 and pfSense 2.4.2.
+
+FullRst was chosen as most promising one: it can be done both by IO as well as
+memory mapped ACPI register. It also puts platform in state closest to cold boot
+from all tested options by staying in S5 for some time.
+
+###### Debian
+
+By default it is impossible to access physical memory space without changes to
+the kernel, however access to IO ports is unrestricted for root user. Thus, to
+perform FullRst type as root:
+
+```
+echo -ne "\xe" | dd of=/dev/port bs=1 count=1 seek=$((0xcf9))
+```
+
+`\xe` is a bitmap for FullRst, RstCmd and SysRst. After this command platform
+enters S5 immediately, so save your work, sync all filesystems etc. using e.g.
+[SysRq](https://askubuntu.com/a/491153), of course do not perform last step
```

Comment:
@krystian-hebel add please the set of commands to trigger sysrq

Reply:
done & tested